mungegithub/mungers/approvers/approvers_test.go

```diff
@@ -0,0 +1,386 @@
+	rootApprovers := sets.NewString("Alice", "Bob")
+	aApprovers := sets.NewString("Art", "Anne")
+	bApprovers := sets.NewString("Bill", "Ben", "Barbara")
+	cApprovers := sets.NewString("Chris", "Carol")
+	dApprovers := sets.NewString("David", "Dan", "Debbie")
+	eApprovers := sets.NewString("Eve", "Erin")
+	edcApprovers := eApprovers.Union(dApprovers).Union(cApprovers)
+	FakeRepoMap := map[string]sets.String{
+		"":        rootApprovers,
+		"a":       aApprovers,
+		"b":       bApprovers,
+		"c":       cApprovers,
+		"a/d":     dApprovers,
+		"a/combo": edcApprovers,
+	}
+	tests := []struct {
+		testName          string
+		filenames         []string
+		currentlyApproved sets.String
+		expectedFiles     []File
+	}{
+		{
+			testName:          "Empty PR",
+			filenames:         []string{},
+			currentlyApproved: sets.NewString(),
+			expectedFiles:     []File{},
+		},
+		{
+			testName:          "Single Root File PR Approved",
+			filenames:         []string{"kubernetes.go"},
+			currentlyApproved: sets.NewString(rootApprovers.List()[0]),
+			expectedFiles:     []File{ApprovedFile{"", sets.NewString(rootApprovers.List()[0])}},
+		},
+		{
+			testName:          "Single File PR in B No One Approved",
+			filenames:         []string{"b/test.go"},
+			currentlyApproved: sets.NewString(),
+			expectedFiles:     []File{UnapprovedFile{"b"}},
+		},
+		{
+			testName:          "Single File PR in B Fully Approved",
+			filenames:         []string{"b/test.go"},
+			currentlyApproved: bApprovers,
+			expectedFiles:     []File{ApprovedFile{"b", bApprovers}},
+		},
+		{
+			testName:          "Single Root File PR No One Approved",
+			filenames:         []string{"kubernetes.go"},
+			currentlyApproved: sets.NewString(),
+			expectedFiles:     []File{UnapprovedFile{""}},
+		},
+		{
+			testName:          "Combo and Other; Neither Approved",
+			filenames:         []string{"a/combo/test.go", "a/d/test.go"},
+			currentlyApproved: sets.NewString(),
+			expectedFiles:     []File{UnapprovedFile{"a/combo"}, UnapprovedFile{"a/d"}},
+		},
+		{
+			testName:          "Combo and Other; Combo Approved",
+			filenames:         []string{"a/combo/test.go", "a/d/test.go"},
+			currentlyApproved: eApprovers,
+			expectedFiles:     []File{ApprovedFile{"a/combo", eApprovers}, UnapprovedFile{"a/d"}},
+		},
+		{
+			testName:          "Combo and Other; Both Approved",
+			filenames:         []string{"a/combo/test.go", "a/d/test.go"},
+			currentlyApproved: edcApprovers.Intersection(dApprovers),
+			expectedFiles:     []File{ApprovedFile{"a/combo", edcApprovers.Intersection(dApprovers)}, ApprovedFile{"a/d", edcApprovers.Intersection(dApprovers)}},
+		},
+		{
+			testName:          "Combo, C, D; Combo and C Approved",
+			filenames:         []string{"a/combo/test.go", "a/d/test.go", "c/test"},
+			currentlyApproved: cApprovers,
+			expectedFiles:     []File{ApprovedFile{"a/combo", cApprovers}, UnapprovedFile{"a/d"}, ApprovedFile{"c", cApprovers}},
+		},
```

Comment:
You have no test where a group of people approve a dir, and another group approves another dir, and maybe a group approves multiple dirs.